### PR TITLE
feat: implement AssetEntryDto JsonLd validator

### LINKED ISSUE #3142: 🐞 bug: Management API accepting empty ("") ids
# Bug Report

## Describe the Bug
the management API allows for null ids

### Expected Behavior
A 400 "Bad Request" Response

### Observed Behavior
The objects are created with `"@id": null`.
The object created cannot be used or deleted.

### Assets 
request POST `/management/v2/assets`
```json
{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@type": "AssetEntryDto",
  "asset": {
    "@id": "",
    "properties": {
      "name": "asset-name"
    }
  },
  "dataAddress": {
    "type": "asset-address-type",
    "keyName": "asset-key-name"
  }
}
```
 Response
```json
{
  "@type": "edc:IdResponseDto",
  "@id": null,
  "edc:createdAt": 1685901622648,
  "@context": {
    "dct": "https://purl.org/dc/terms/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "dcat": "https://www.w3.org/ns/dcat/",
    "odrl": "http://www.w3.org/ns/odrl/2/",
    "dspace": "https://w3id.org/dspace/v0.8/"
  }
}
```

### Policies
Request POST `/management/v2/policydefinitions
```json
{
  "@context": {
    "edc": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@id": "",
  "policy": {
    "odrl:permission": [
      {
        "odrl:action": "USE",
        "odrl:constraint": [
          {
            "odrl:leftOperand": "BusinessPartnerNumber",
            "odrl:operator": "EQ",
            "odrl:rightOperand":  "BPN"
          }]
      }
    ],
    "odrl:prohibition": [],
    "odrl:obligation": []
  }
}
```
Response
```json
{
  "@type": "edc:IdResponseDto",
  "@id": null,
  "edc:createdAt": 1685902093330,
  "@context": {
    "dct": "https://purl.org/dc/terms/",
    "edc": "https://w3id.org/edc/v0.0.1/ns/",
    "dcat": "https://www.w3.org/ns/dcat/",
    "odrl": "http://www.w3.org/ns/odrl/2/",
    "dspace": "https://w3id.org/dspace/v0.8/"
  }
}
```

## Context Information
_Add any other context about the problem here._

- Used version: milestone 9

### PR DESCRIPTION
**Pull request by ndr-brt**
## What this PR changes/adds

Implement `AssetEntryDto` validator.

## Why it does that

validation

## Further notes

- the validator has been implemented as a "rule" engine, and an instance can be built fluently, see `AssetEntryDtoValidator` for an example
- some further work needs to be done, e.g. use "problem builder pattern" for violations

## Linked Issue(s)

Closes #3142 

_Please be sure to take a look at the [contributing guidelines](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and our [etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md)._

### CODE REVIEW

**Review by jimmarino — APPROVED** [review]
Cool stuff!

File: extensions/control-plane/api/management-api/asset-api/src/main/java/org/eclipse/edc/connector/api/management/asset/validation/AssetEntryDtoValidator.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ */
+public class AssetEntryDtoValidator {
+
+    public static Validator<JsonObject> instance() {
+        return JsonObjectValidator.newValidator()
+                .verify(EDC_ASSET_ENTRY_DTO_ASSET, MandatoryField::new)
+                .verifyObject(EDC_ASSET_ENTRY_DTO_ASSET, v -> v
+                        .verify(ID, OptionalIdNotBlank::new)
+                        .verify(EDC_ASSET_PROPERTIES, MandatoryField::new)
+                        .verify("", path -> new AssetPropertiesUniqueness())
```
Comment (wolf4ood) [inline extensions/control-plane/api/management-api/asset-api/src/main/java/org/eclipse/edc/connector/api/management/asset/validation/AssetEntryDtoValidator.java:43]: If this validator is working on the entire object `v` in this case i would make another API that hides the passing of `""` and only taking  the validator provider function

**Review by wolf4ood — APPROVED** [review]
Neat stuff :rocket:  just a minor nit

File: extensions/data-plane/data-plane-client/src/test/java/org/eclipse/edc/connector/core/validator/JsonObjectValidatorRegistryImplTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import static jakarta.json.Json.createObjectBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.junit.assertions.AbstractResultAssert.assertThat;
+import static org.eclipse.edc.validator.spi.Violation.violation;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class JsonObjectValidatorRegistryImplTest {
```
Comment (ronjaquensel) [inline extensions/data-plane/data-plane-client/src/test/java/org/eclipse/edc/connector/core/validator/JsonObjectValidatorRegistryImplTest.java:31]: I think this test landed in the wrong module, at least the GitHub overview is showing it under `extensions/data-plane/data-plane-client` which doesn't seem right

**Review by ronjaquensel — APPROVED** [review]

**Pull request merged**